envoy/buffer/buffer.h

```diff
@@ -458,6 +458,31 @@ class Instance {
     writeInt<ByteOrder::BigEndian, T, Size>(value);
   }
 
+  /**
+   * Copy multiple arguments to the buffer. In order to improve performance, all the arguments in
+   * each call will be stored in contiguous memory. So this method is mainly used when a large
+   * amount of fine-grained data needs to be written to the buffer. The type of arguments must be
+   * automatically converted to `absl::string_view`.
+   * @param args A set of arguments with variable length and type. The arguments must be guaranteed
+   * to be safely converted to `absl::string_view`.
+   * @return The total size of the data copied to the buffer.
+   */
+  template <class... Args> size_t addFragments(Args&&... args) {
+    size_t total_size_to_write = 0;
+    if constexpr (sizeof...(args) > 0) {
```

Comment:
Would it be possible to require at least 2 arguments?

Single argument cases are better served by add.  0 argument cases are not a thing.

Reply:
This is possible. But is it really necessary? We can recommend some ways to use it, but keeping the ability to handle single parameters in addFragments is not a bad thing, right?

Reply:
I'd prefer requiring 2 or more args for this, per Antonio's comments. I think you can also use static_assert to check that.

Reply:
OK. Since both of you suggested so.